### PR TITLE
Ruby Program to reverse an array

### PR DESCRIPTION
**Pull request by shubhamnandanwar**

### CODE REVIEW

File: reverse_array.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    end
+  end
+end
+
+puts "Enter the size of the array\n"
+n = gets.chomp.to_i
+puts "\nEnter the Array Elements"
+input_array=Array.new(n)
+
+for i in (0...n)
```
Comment (ndashore) [inline reverse_array.rb]: Do it without for loop.

File: reverse_array.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
+class Array
+
+  def reverse_n
+    self[0, self.length / 2].each_with_index do |input_element,index|
+      self[index], self[self.length - 1 - index] = self[self.length - 1 - index], self[index]
```
Comment (ndashore) [inline reverse_array.rb]: Why are you getting length again and again?

File: reverse_array.rb (hunk 1 above)
Comment (shubhamnandanwar) [reply reverse_array.rb]: Can I use inbuilt function then?

Comment (ndashore) [reply reverse_array.rb]: I said do it without `for` loop. I told you that you should not use `for` and `while` loops.

File: reverse_array.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    end
+  end
+end
+
+puts "Enter the size of the array\n"
+n = gets.chomp.to_i
+puts "\nEnter the Array Elements"
+input = Array.new(n)
+
+(0...n).each do |index|
```
Comment (ndashore) [inline reverse_array.rb]: Can you give a try to `times` loop?

File: reverse_array.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
+class Array
+  def reverse_n
+    self[0, self.length / 2].each_with_index do |input_element,index|
+      self[index], self[self.length - 1 - index] = self[self.length - 1 - index], self[index]
```
Comment (ndashore) [inline reverse_array.rb]: Still calling `self.length` 3 times.

File: reverse_array.rb [hunk 5]
```diff
@@ -0,0 +1,21 @@
+class Array
+  def reverse_n
+  	length = self.length
```
Comment (ndashore) [inline reverse_array.rb]: indentation?

**Pull request merged**